### PR TITLE
[FWD-396] Support transmission ids

### PR DESCRIPTION
**Pull request by m-sasha**

### CODE REVIEW

**Comment by tomerg-flytrex** [comment]
Unless you say otherwise, I'm going to not review anything under java-gen.

**Comment by m-sasha** [comment]
This is more of an "FYI" PR.

File: buildSrc/src/main/java/io/dronefleet/mavlink/generator/MessageGenerator.java [hunk 1]
```diff
@@ -222,7 +235,23 @@ public TypeSpec generate() {
...
-                .addMethod(generateToString())
-                .build();
+                .addMethod(generateToString());
+
+        boolean hasTransmissionId = fields.stream()
+                .anyMatch(field -> "tid".equals(field.getName()) && "uint8_t".equals(field.getType()));
+        if (hasTransmissionId){
+            builder.addSuperinterface(ParameterizedTypeName.get(HAS_TRANSMISSION_ID, className));
+            builder.addMethod(MethodSpec
+                    .methodBuilder("withTransmissionId")
```
Comment (tomerg-flytrex) [inline buildSrc/src/main/java/io/dronefleet/mavlink/generator/MessageGenerator.java:245]: What's the use of `withTransmissionId`? When are we going to copy a message and just change the tid?

Comment (m-sasha) [reply buildSrc/src/main/java/io/dronefleet/mavlink/generator/MessageGenerator.java:245]: The tids will managed at a low level (in MavlinkConnectionWrapper), so it needs the ability to change it in `send()`.

**Comment by tomerg-flytrex** [comment]
I don't think MC-396 is the right ticket for this commit

**Comment by m-sasha** [comment]
Right. It should be FWD.

File: buildSrc/src/main/java/io/dronefleet/mavlink/generator/MessageGenerator.java (hunk 1 above)
Comment (tomerg-flytrex) [reply buildSrc/src/main/java/io/dronefleet/mavlink/generator/MessageGenerator.java:245]: Keep in mind it means allocating every message twice.

Comment (m-sasha) [reply buildSrc/src/main/java/io/dronefleet/mavlink/generator/MessageGenerator.java:245]: Not too worried, we're not sending that many.

**Review by tomerg-flytrex — APPROVED** [review]

**Pull request closed without merging**